manifests/09-prometheus-rules.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    name: image-registry-operator-alerts
+  name: image-registry-operator-alerts
+  namespace: openshift-image-registry
+spec:
+  groups:
+  - name: ImageRegistryOperator
+    rules:
+    - alert: ImageRegistryMissingStorageConfiguration
+      expr: max_over_time(image_registry_operator_storage_configured[48h]) == 0
+      annotations:
+        message: |
+          Image Registry misses a backend storage configuration. This alert
+          will auto resolve once Image Registry is provided with a valid
+          backend configuration.
+    - alert: ImageRegistryDegraded
+      expr: image_registry_operator_status_degraded == 1
+      annotations:
+        message: |
+          Image Registry could not be deployed and is in Degraded status.
+    - alert: ImageRegistryStorageBackendReconfigured
+      expr: increase(image_registry_operator_storage_backend_created[30m]) > 0
```

Comment:
at some point a configuration will have to be made no? This will fire even on initial configuration

Reply:
I haven't seen this issue, I guess that is because Prometheus generally takes some time to actually query the metrics endpoint. I have refactored the code to support this scenario. Thanks for catching this, PTAL.